src/main/webapp/2Dscript.js

```diff
@@ -1,56 +1,102 @@
 // Gobal variables.
 let map;
+let maps = {}; // Cache for map elements that have been loaded.
+let datas = {}; // Cache for data elements that have been loaded.
 let pose;
-let runId;
-let dataType;
+let data;
+let id;
+let type;
+let currentLat;
+let currentLong;
+// Define bounding box around the different data sets to quickly determined
+// whether they should be included in the calculation or not
 
+/**
+ * A Point object with a lat and lng.
+ * @typedef {Object<number, number>} Point
+ * @property {number} lat The lat of the point.
+ * @property {number} lng The lng of the point.
+ */
+let flightPath;
+// Define bounding box around the different data sets to quickly determined whether they should be included in the calculation or not
 fetchData();
 /**
- * Fetchs pose data from the RunInfo servlet,
- * it is an asynchronous call requiring initMap() to be
- * called after the data is fully loaded.
+ * Fetchs pose data from the RunInfo servlet, it is an asynchronous call
+ * requiring initMap() to be called after the data is fully loaded.
  */
 function fetchData() {
   const queryString = window.location.search;
   const urlParams = new URLSearchParams(queryString);
-  runId = urlParams.get('id');
-  dataType = urlParams.get('dataType');
-  fetch('/getrun?id=' + runId + '&dataType=' + dataType)
-      .then((response) => response.json())
-      .then((data) => pose = data)
-      .then(() => initMap());
+  id = urlParams.get('id');
+  type = urlParams.get('dataType');
+  fetch('/getrun?id=' + id + '&dataType=' + type)
+    .then(response => response.json())
+    .then(data => pose = data)
+    .then(() => { return fetch('\data') })
+    .then(response => response.json())
+    .then(json => data = json)
+    .then(() => {
+      initMap();
+    })
 }
 
 /**
- * Embeds the Google map interface within the html
- * along with drawing the pose trajectory and adding a button link.
+ * Embeds the Google map interface within the html along with drawing the pose
+ * trajectory and adding a button link.
  */
 function initMap() {
-  // Intitalizes the map.
+  // Occurs when google map api calls all promises, which is getting this
+  // function called somehow, according to stack trace.
+  if (pose === undefined) {
+    return;
+  }
+
+  console.log('initMap called');
+  const poseCoordinates = [];
+  for (let i = 0; i < pose.length; i++) {
```

Comment:
Can you use `for (const point of pose)`?

https://google.github.io/styleguide/jsguide.html#features-for-loops